canal/circuit.py

```
@@ -579,6 +581,9 @@ def __should_add_config(self):
         return False
 
     def finalize(self):
+        if self.finalized:
+            return
```

instead of `return`, let's raise an error

---

Sure. Do you know what happened to the `BitVector` that broke every tests?